capstone/src/main/java/com/google/sps/servlets/getNYTDataServlet.java

```diff
@@ -31,27 +31,29 @@
 /*Servlet that gets COVID count data from NYT*/
 @WebServlet("/getCountyData")
 public class getNYTDataServlet extends HttpServlet {
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String link = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/live/us-counties.csv";
-    URL url = new URL(link);
-    HttpURLConnection url_connect = (HttpURLConnection) url.openConnection();
-    url_connect.setRequestMethod("GET");
-	int code = url_connect.getResponseCode();
-    List<String> county_data = new ArrayList<String>();
-    if(code == HttpURLConnection.HTTP_OK){ 
-	  BufferedReader reader = new BufferedReader(new InputStreamReader(url_connect.getInputStream()));
-      String input = reader.readLine();
-      while(input != null) {
-        input = reader.readLine();
-        county_data.add(input);
-	  }
-	  reader.close();
-	}else{
-      county_data.add("error");
-	}
-    String json = new Gson().toJson(county_data);
-    response.setContentType("application/json");
-    response.getWriter().println(json);
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        List<String> county_data = new ArrayList <String>();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        try {
+            String link = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/live/us-counties.csv";
+            URL url = new URL(link);
```

Comment:
Move any lines unrelated to IO outside of the try block if possible. Try blocks execute a little bit slower in some situations.